src/config.rs

```diff
@@ -23,32 +22,63 @@ const DEFAULT_SYMLINK_ROOT: &'static str = "/var/run/gpio";
 #[derive(Debug, PartialEq, Clone)]
 pub struct Direction(pub sysfs_gpio::Direction);
 
+#[derive(Deserialize, Debug)]
+#[serde(remote = "sysfs_gpio::Direction")]
+pub enum DirectionDef {
+    #[serde(rename = "in")]
+    In,
+    #[serde(rename = "out")]
+    Out,
+    #[serde(rename = "high")]
+    High,
+    #[serde(rename = "low")]
+    Low,
+}
+
 impl From<sysfs_gpio::Direction> for Direction {
     fn from(e: sysfs_gpio::Direction) -> Self {
         Direction(e)
     }
 }
 
-#[derive(Debug, Clone, PartialEq)]
+#[derive(Debug, Clone, PartialEq, Deserialize)]
 pub struct PinConfig {
     pub num: u64,
+    #[serde(default = "default_direction")]
+    #[serde(with = "DirectionDef")]
     pub direction: sysfs_gpio::Direction,
+    #[serde(default)]
     pub names: BTreeSet<String>,
+    #[serde(default = "bool_true")]
     pub export: bool,
+    #[serde(default)]
     pub active_low: bool,
 }
 
-#[derive(Clone, Debug, Default)]
+fn default_direction()-> sysfs_gpio::Direction {
+    sysfs_gpio::Direction::In
+}
+
+fn bool_true()-> bool {
```

Comment:
Couple of these things have minor formatting issues; `rustfmt` can sort these out.